### DISCUSSION TITLE
v1.18.1 (4689926) — This osTicket version is no longer supported. Please consider upgrading

### DISCUSSION THREAD

**Discussion by Kamrang-aja** [osTicket/osTicket#6678] [Q&A] [answered]
Hello, I hope you help me, I get a display on the version status which should be Up to date but it is still detected as an old version (This osTicket version is no longer supported. Please consider upgrading).

Another person said that:

- May not be able to check the version online, but my server can reach the internet and there should be no problems with this
- Or maybe there is a problem with system.inc.php for the getLatestVersion variable, I don't know if it is true and I don't dare to change what is in system.inc.php.

![4](https://github.com/osTicket/osTicket/assets/72611915/42e7608a-11f0-4f13-9169-a3eec2a0ca36)

**Comment by JediKev** [comment]
@Kamrang-aja 

Sounds like your server is not finding the DNS records the system needs to determine the latest version. I would suggest doing a DNS TXT record lookup on the `core.updates.osticket.com` domain from the server's cli to verify it can reach it and get an appropriate answer. Here is an example using `dig`:
```cli
dig -t txt core.updates.osticket.com
```

You should get a response that includes an answer like this:
```cli
;; ANSWER SECTION:
core.updates.osticket.com. 7200	IN	TXT	"v=1; m=1.10; V=v1.10.7; c=dca01e1; s=98ad7d55"
core.updates.osticket.com. 7200	IN	TXT	"v=1; m=1.12; V=v1.12.6; c=e351ba5; s=00c949a6"
core.updates.osticket.com. 7200	IN	TXT	"v=1; m=1.14; V=v1.14.8; c=7398e90; s=4bd47d94"
core.updates.osticket.com. 7200	IN	TXT	"v=1; m=1.15; V=v1.15.8; c=0daca43; s=add62892"
core.updates.osticket.com. 7200	IN	TXT	"v=1; m=1.16; V=v1.16.6; c=30ef1cc; s=c37e1656"
core.updates.osticket.com. 7200	IN	TXT	"v=1; m=1.17; V=v1.17.5; c=7c20036; s=83a22ba2"
core.updates.osticket.com. 7200	IN	TXT	"v=1; m=1.18; V=v1.18.1; c=4689926; s=5fb92bef"
```

**Note:**
For future reference, please use [the Forum](https://forum.osticket.com) for any issues like this (unless you can confirm it's an actual bug in the code). If it's an actual confirmed bug then you would need to [open an Issue](https://github.com/osTicket/osTicket/issues/new) rather than creating a Discussion.

Cheers.

**Reply by Kamrang-aja** [reply]
Hi @JediKev,

Thank you for your response, I can reach it as below:

```
# dig -t txt core.updates.osticket.com

; <<>> DiG 9.16.48-Debian <<>> -t txt core.updates.osticket.com
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 41208
;; flags: qr rd ra; QUERY: 1, ANSWER: 7, AUTHORITY: 4, ADDITIONAL: 3

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 4096
; COOKIE: ec789d5106198569136cd08165f9f044f865fc9cb09d107a (good)
;; QUESTION SECTION:
;core.updates.osticket.com.	IN	TXT

;; ANSWER SECTION:
core.updates.osticket.com. 6774	IN	TXT	"v=1; m=1.12; V=v1.12.6; c=e351ba5; s=00c949a6"
core.updates.osticket.com. 6774	IN	TXT	"v=1; m=1.10; V=v1.10.7; c=dca01e1; s=98ad7d55"
core.updates.osticket.com. 6774	IN	TXT	"v=1; m=1.17; V=v1.17.5; c=7c20036; s=83a22ba2"
core.updates.osticket.com. 6774	IN	TXT	"v=1; m=1.18; V=v1.18.1; c=4689926; s=5fb92bef"
core.updates.osticket.com. 6774	IN	TXT	"v=1; m=1.14; V=v1.14.8; c=7398e90; s=4bd47d94"
core.updates.osticket.com. 6774	IN	TXT	"v=1; m=1.15; V=v1.15.8; c=0daca43; s=add62892"
core.updates.osticket.com. 6774	IN	TXT	"v=1; m=1.16; V=v1.16.6; c=30ef1cc; s=c37e1656"

;; AUTHORITY SECTION:
updates.osticket.com.	47517	IN	NS	ns-467.awsdns-58.com.
updates.osticket.com.	47517	IN	NS	ns-603.awsdns-11.net.
updates.osticket.com.	47517	IN	NS	ns-1237.awsdns-26.org.
updates.osticket.com.	47517	IN	NS	ns-1835.awsdns-37.co.uk.

;; ADDITIONAL SECTION:
ns-467.awsdns-58.com.	18309	IN	A	205.251.193.211
ns-603.awsdns-11.net.	57227	IN	A	205.251.194.91

;; Query time: 3 msec
;; SERVER: 202.75.17.2#53(202.75.17.2)
;; WHEN: Wed Mar 20 03:06:28 WIB 2024
;; MSG SIZE  rcvd: 657
```

And thank you for your short note.

**Reply by JediKev** [reply]
@Kamrang-aja 

Your DNS is wayyy behind. Your DNS shows v1.16.6 being the latest when in reality v1.18.1 is. I mean just compare my TXT answer to yours; you can see the clear difference. You need to flush your DNS or something. If all else fails contact your server admin for further assistance or look at various guides online.

Cheers.

**Reply by Kamrang-aja** [reply]
@JediKev 

I think the TXT answer given is random, here is a comparison of the dig results and here are the results:

Dig 1st:
``` 
;; ANSWER SECTION:
core.updates.osticket.com. 6774	IN	TXT	"v=1; m=1.12; V=v1.12.6; c=e351ba5; s=00c949a6"
core.updates.osticket.com. 6774	IN	TXT	"v=1; m=1.10; V=v1.10.7; c=dca01e1; s=98ad7d55"
core.updates.osticket.com. 6774	IN	TXT	"v=1; m=1.17; V=v1.17.5; c=7c20036; s=83a22ba2"
core.updates.osticket.com. 6774	IN	TXT	"v=1; m=1.18; V=v1.18.1; c=4689926; s=5fb92bef"  <<<
core.updates.osticket.com. 6774	IN	TXT	"v=1; m=1.14; V=v1.14.8; c=7398e90; s=4bd47d94"
core.updates.osticket.com. 6774	IN	TXT	"v=1; m=1.15; V=v1.15.8; c=0daca43; s=add62892"
core.updates.osticket.com. 6774	IN	TXT	"v=1; m=1.16; V=v1.16.6; c=30ef1cc; s=c37e1656"
```

Dig 2nd:
```
;; ANSWER SECTION:
core.updates.osticket.com. 3341	IN	TXT	"v=1; m=1.12; V=v1.12.6; c=e351ba5; s=00c949a6"
core.updates.osticket.com. 3341	IN	TXT	"v=1; m=1.16; V=v1.16.6; c=30ef1cc; s=c37e1656"
core.updates.osticket.com. 3341	IN	TXT	"v=1; m=1.17; V=v1.17.5; c=7c20036; s=83a22ba2"
core.updates.osticket.com. 3341	IN	TXT	"v=1; m=1.14; V=v1.14.8; c=7398e90; s=4bd47d94"
core.updates.osticket.com. 3341	IN	TXT	"v=1; m=1.15; V=v1.15.8; c=0daca43; s=add62892"
core.updates.osticket.com. 3341	IN	TXT	"v=1; m=1.10; V=v1.10.7; c=dca01e1; s=98ad7d55"
core.updates.osticket.com. 3341	IN	TXT	"v=1; m=1.18; V=v1.18.1; c=4689926; s=5fb92bef"  <<<
```

**Comment by JediKev** [comment]
@Kamrang-aja 

That is so weird; they should be in order. Nonetheless, that’s my bad for overlooking that.

The next step would be to dump the contents of the results of the DNS lookup within the code to see if PHP is failing the DNS lookup for some reason. Go to [this line](https://github.com/osTicket/osTicket/blob/develop/include/class.osticket.php#L420) and below it on a new line add `var_dump($records);die;`. Then revisit the information page and see the output.

Cheers.

**Reply by Kamrang-aja** [reply]
@JediKev 

Is it like this?

```
     function getLatestVersion($product='core', $major=null) {
        $records = dns_get_record($product.'.updates.osticket.com', DNS_TXT);
        var_dump($records);die;
        if (!$records)
            return false;

```

Output :
![5](https://github.com/osTicket/osTicket/assets/72611915/838be25b-4273-4454-b0ee-5b195ce6620b)

**Reply by JediKev** [reply]
@Kamrang-aja 

Precisely, as you can see the DNS lookup from PHP (via the native `dns_get_record` function) is failing to return results. Do you see any errors in your PHP error logs? Are you able to contact your hosting provider to see if they have any restrictions/limitations on PHP?

Cheers.

**Reply by Kamrang-aja** [reply]
@JediKev 

For your information, I use Nginx, php8.1-fpm and php8.1, and I don't get any PHP error logs. I have also tried reinstalling the suggested PHP extension but the problem is still the same.

**Reply by JediKev** [reply]
@Kamrang-aja 

Just to note, NGINX is not fully supported out of the box as it requires the manual addition of NGINX rules that we don’t have available (there are some rules available online though - even on the NGINX website). With this being said that should have nothing to do with this specific issue; just simply letting you know.

I have tested this locally, on a hosted server, and via the [OnlinePHP.io Sandbox](https://onlinephp.io/) (`var_dump(dns_get_record('core.updates.osticket.com', DNS_TXT));`). All tests returned the expected results. This basically means the issue you are experiencing is specific to your server/setup. I would highly suggest reaching out to your hosting provider and/or ISP to dig deeper into the PHP DNS issues.

Cheers.

**Reply by Kamrang-aja** [reply, marked as answer]
@JediKev 

I have found the problem, it is true that you said there is a problem with DNS, now I changed it to Google DNS 8.8.8.8, the problem is resolved.

![image](https://github.com/osTicket/osTicket/assets/72611915/1280a75d-2a2e-4b6e-8f09-428fb69e0f17)

I am very grateful for your help.

**Reply by Kamrang-aja** [reply]
@JediKev 

Additional information, I'm still confused about why the dig I did under my DNS produces TXT randomly, different from the dig results if I use Google's DNS 8.8.8.8 which is sequential as you said.

```
;; ANSWER SECTION:
core.updates.osticket.com. 7080	IN	TXT	"v=1; m=1.10; V=v1.10.7; c=dca01e1; s=98ad7d55"
core.updates.osticket.com. 7080	IN	TXT	"v=1; m=1.12; V=v1.12.6; c=e351ba5; s=00c949a6"
core.updates.osticket.com. 7080	IN	TXT	"v=1; m=1.14; V=v1.14.8; c=7398e90; s=4bd47d94"
core.updates.osticket.com. 7080	IN	TXT	"v=1; m=1.15; V=v1.15.8; c=0daca43; s=add62892"
core.updates.osticket.com. 7080	IN	TXT	"v=1; m=1.16; V=v1.16.6; c=30ef1cc; s=c37e1656"
core.updates.osticket.com. 7080	IN	TXT	"v=1; m=1.17; V=v1.17.5; c=7c20036; s=83a22ba2"
core.updates.osticket.com. 7080	IN	TXT	"v=1; m=1.18; V=v1.18.1; c=4689926; s=5fb92bef"
```
